src/App.tsx

```diff
@@ -1,54 +1,83 @@
-import React from 'react';
 import './App.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { debounce } from 'lodash';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownItem } from './Components/DropdownItem';
+
+function filterPeopleByQuery(peopleToFilter: Person[], query: string) {
+  const queryToLowerCase = query.toLowerCase().trim();
+
+  return peopleToFilter
+    .filter(({ name }) => name.toLowerCase().includes(queryToLowerCase));
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [hasFocus, setHasFocus] = useState(false);
+
+  const handleFocus = () => {
+    setHasFocus(true);
+  };
+
+  const filteredPeople: Person[] = useMemo(() => {
+    return filterPeopleByQuery(peopleFromServer, appliedQuery);
+  }, [appliedQuery]);
+
+  const applyQuery
+  = useCallback(debounce(setAppliedQuery, 1000), [appliedQuery]);
```

Comment:
This `useCallback` is not needed + a magic value of `1000` (please re-check the task)